polygeist/tools/cgeist/driver.cc

```diff
@@ -305,8 +303,16 @@ static int canonicalize(mlir::MLIRContext &Ctx,
       OptPM.addPass(mlir::createLoopInvariantCodeMotionPass());
     OptPM.addPass(polygeist::createRaiseSCFToAffinePass());
     OptPM.addPass(polygeist::replaceAffineCFGPass());
-    if (ScalarReplacement)
+    if (ScalarReplacement) {
+      // Perform AffineScalarReplacementPass on func::FuncOp nested in another
+      // region, e.g., gpu.module.
+      mlir::OpPassManager &nestPM = PM.nestAny();
+      nestPM.addNestedPass<func::FuncOp>(
+          mlir::createAffineScalarReplacementPass());
+      // Perform AffineScalarReplacementPass on func::FuncOp directly under
+      // builtin.module.
       PM.addNestedPass<func::FuncOp>(mlir::createAffineScalarReplacementPass());
```

Comment:
We discussed this offline, would make sense to create a lambda function to make it easier in the future to add more function passes that can operate on a `func::FuncOp`  whether it is nested directly into a builtin.module or a gpu.module.

Reply:
done.